examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,578 @@
+# evaluated with :class:`~sklearn.model_selection.GridSearchCV`. In this case
+# we would be running our statistical test multiple times, which leads us to
+# the `multiple comparisons problem
+# <https://en.wikipedia.org/wiki/Multiple_comparisons_problem>`_.
+#
+# There are many possible ways to tackle this problem, but a standard approach
+# is to apply a `Bonferroni correction
+# <https://en.wikipedia.org/wiki/Bonferroni_correction>`_. Bonferroni can be
+# computed by multiplying the p-value by the number of comparisons we are
+# testing.
+#
+# Let's compare the performance of the models using the corrected t-test:
+
+from itertools import combinations
+from math import factorial
+
+n_comparisons = (
+    factorial(len(model_scores))
+    / (factorial(2) * factorial(len(model_scores) - 2))
+)
+pairwise_t_test = []
+
+for model_i, model_k in combinations(range(len(model_scores)), 2):
+    model_i_scores = model_scores.iloc[model_i].values
+    model_k_scores = model_scores.iloc[model_k].values
+    differences = model_i_scores - model_k_scores
+    t_stat, p_val = compute_corrected_ttest(
+        differences, n, df, n_train, n_test
+    )
+    p_val *= n_comparisons  # implement Bonferroni correction
+    # Bonferroni can output p-values higher than 1
+    p_val = 1 if p_val > 1 else p_val
+    pairwise_t_test.append(
+        [model_scores.index[model_i], model_scores.index[model_k],
+         t_stat, p_val]
+    )
+
+pairwise_comp_df = pd.DataFrame(
+    pairwise_t_test,
+    columns=['model_1', 'model_2', 't_stat', 'p_val']
+).round(3)
+pairwise_comp_df
+
+# %%
+# We observe that after correcting for multiple comparisons, the only model
+# that significantly differs from the others is `'2_poly'`.
+# `'rbf'`, the model ranked first by
+# :class:`~sklearn.model_selection.GridSearchCV`, does not significantly
+# differ from `'linear'` or `'3_poly'`.
+
+# %%
+# Pairwise comparison of all models: Bayesian approach
+# ----------------------------------------------------
+#
+# When using Bayesian estimation to compare multiple models, we don't need to
+# correct for multiple comparisons (for reasons why see [4]_).
+#
+# We can carry out our pairwise comparisons the same way as in the first
+# section:
+
+pairwise_bayesian = []
+
+for model_i, model_k in combinations(range(len(model_scores)), 2):
+    model_i_scores = model_scores.iloc[model_i].values
+    model_k_scores = model_scores.iloc[model_k].values
+    differences = model_i_scores - model_k_scores
+    t_post = t(
+        df, loc=np.mean(differences),
+        scale=correct_std(differences, n, n_train, n_test)
+    )
+    left_prob = t_post.cdf(rope_interval[0])
+    right_prob = t_post.sf(rope_interval[1])
+    rope_prob = t_post.cdf(rope_interval[1]) - t_post.cdf(rope_interval[0])
+
+    pairwise_bayesian.append([left_prob, right_prob, rope_prob])
+
+pairwise_bayesian_df = (pd.DataFrame(
+    pairwise_bayesian,
+    columns=['worse_prob', 'better_prob', 'rope_prob']
```

Comment:
isn't better_prob equal to 1 - right_prob?

Seems like it's set to right_prob ATM

EDIT nvm, you use `sf`. Maybe still use `1 - cdf` since this is what was done above. Also I would just name `left_prob` and `right_prob` `worse_prob` and `better_prob` directly